plugin/pkg/dnstest/server.go

```diff
@@ -23,18 +23,28 @@ func NewServer(f dns.HandlerFunc) *Server {
 	ch1 := make(chan bool)
 	ch2 := make(chan bool)
 
-	l, _ := net.Listen("tcp", ":0")
-	if l == nil {
-		return nil
+	s1 := &dns.Server{} // udp
+	s2 := &dns.Server{} // tcp
+
+	for i := 0; i < 1000; i++ { // 1000 attempts
```

Comment:
Can we make it, 5 ? 1000 looks like very much an overkill?

Reply:
It depends on how much free port system has. Yes, 5 attempts may be enough. On the other hand I don't see any drawback with 1000 attempts - it won't take much time. If you prefer 5 I can change this